lib/api-utils.ts

```diff
@@ -0,0 +1,7 @@
+export function checkError(response) {
+  if (response && response.data && response.data.error_code !== 0) {
```

Comment:
We could split this out into multiple branches...that way if there's no response a different error occurs

Reply:
probably not a bad idea. i also saw on the tplink website in some of the camera code that there is some other strange logic they used. Not sure if we need it here, but:

https://d2baw28zdve8g3.cloudfront.net/ssl/tpcamera-admin/js/ipc-business-libs.js

```js
                var errCodeStrIndex = inputArgs["errCodeStrIndex"] || "errorCode";
                var noErrorCode = inputArgs["noErrorCode"] || 0;
                var defaultErrorCode = inputArgs["defaultErrorCode"] || -1;

                if (response[errCodeStrIndex] == noErrorCode) {
...